### PR TITLE
[Typing][PEP585 Upgrade][BUAA][131-150] Use standard collections for type hints for 19 files in `test/`

### PR DESCRIPTION
**Pull request by Whsjrczr**
### PR Category

User Experience

### PR Types

Others

### Description

修改了以下19个文件：
        modified: test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py
        modified: test/deprecated/legacy_test/auto_parallel_op_test.py
        modified: test/dygraph_to_static/test_dynamic_shape_infermeta.py
        modified: test/dygraph_to_static/test_typehint.py
        modified: test/dygraph_to_static/test_typing.py
        modified: test/ipu/op_test_ipu.py
        modified: test/ir/inference/auto_scan_test.py
        modified: test/ir/inference/program_config.py
        modified: test/ir/inference/test_trt_convert_activation.py
        modified:   test/ir/inference/test_trt_convert_affine_channel.py
        modified:   test/ir/inference/test_trt_convert_anchor_generator.py
        modified:   test/ir/inference/test_trt_convert_arg_max.py
        modified:   test/ir/inference/test_trt_convert_arg_min.py
        modified:   test/ir/inference/test_trt_convert_argsort.py
        modified:   test/ir/inference/test_trt_convert_assign.py
        modified:   test/ir/inference/test_trt_convert_batch_norm.py
        modified:   test/ir/inference/test_trt_convert_bilinear_interp_v2.py
        modified:   test/ir/inference/test_trt_convert_bitwise_and.py
        modified:   test/ir/inference/test_trt_convert_bitwise_not.py
注：test/dygraph_to_static/check_approval.py报错为U036

### CODE REVIEW

File: test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 
 import math
-from typing import Optional, Tuple
+from typing import Optional
```
Comment (SigureMo) [inline test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py]: Optional 保留的理由是？

File: test/ir/inference/test_trt_convert_bitwise_not.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def generate_input1(dims, batch, attrs: List[Dict[str, Any]]):
 
     def sample_predictor_configs(
         self, program_config
-    ) -> (paddle_infer.Config, List[int], float):
+    ) -> (paddle_infer.Config, list[int], float):
```
Comment (SigureMo) [inline test/ir/inference/test_trt_convert_bitwise_not.py]:
```suggestion
    ) -> tuple[paddle_infer.Config, list[int], float]:
```

这里是明显错误的，需要修改

别说一个就改一个，其他的也需要改

**Review by SigureMo — COMMENTED** [review]
一个文件都没用 PEP 563？任务描述是怎么看的？没有 PEP 563 怎么升级 PEP 585？

描述里说了先改一个文件，等 approve 后再继续改，然后你直接改了 19 个，是多么自信啊？

所有 PR 都需要重新确认是否使用了 PEP 563，不然 3.8 会挂！

File: test/ir/inference/test_trt_convert_bitwise_not.py (hunk 2 above)
Comment (Whsjrczr) [reply test/ir/inference/test_trt_convert_bitwise_not.py]: 抱歉，已全部修改

**Review by SigureMo — DISMISSED** [review]
LGTMeow

**Review by SigureMo — APPROVED** [review]

**Pull request merged**